dmoj/problem.py

```diff
@@ -226,6 +226,14 @@ def __init__(self, batch_no, config, problem, cases):
         if any(isinstance(case, BatchedTestCase) for case in self.batched_cases):
             raise InvalidInitException('nested batches')
         self.problem = problem
+        if any(requirement >= batch_no for requirement in self.requirements):
+            raise InvalidInitException('requirements depends on non-earlier batch')
+        if any(requirement < 1 for requirement in self.requirements):
+            raise InvalidInitException('requirements must be positive integers')
+
+    @property
+    def requirements(self):
+        return self.config.get('requirements', [])
```

Comment:
You can add `'requirements;: []` to the default `ProblemConfig` and not need this property.

Reply:
Seems to not work, since this is a batch not a ProblemConfig.

Reply:
You can do `self.requirements = config.requirements`, as is done for points.

Reply:
I'm not thoroughly convinced this is sensible, because this would do weird things if someone specified at the top level, but I've pushed this anyway.

Reply:
The behavior is identical to your original code, because of how `get` is implemented on `ConfigNode`.